lib/ddr/models/has_struct_metadata.rb

```diff
@@ -11,6 +11,47 @@ module HasStructMetadata
 
         has_attributes :file_group, :file_use, :order,
                        datastream: "structMetadata", multiple: false
+
+        after_create :assign_struct_metadata!
+      end
+
+      def assign_struct_metadata!
+        self.file_use = default_file_use unless self.file_use
+        self.order = default_order unless self.order
+        self.file_group = default_file_group unless self.file_group
+        self.save!
+      end
+
+      private
+
+      def default_file_use
+        if self.has_content?
+          case self.content_type
+          when 'image/tiff'
+            'master'
+          else
+            'reference'
+          end
+        end
+      end
+
+      def default_order
+        siblings.size + 1
+      end
+
+      def default_file_group
+        self.identifier.first if self.has_content? && self.describable? && self.identifier.present?
```

Comment:
At this point, do we need self.describable?  I think we could simplify to:

```ruby
identifier.first if has_content?
```

since `identifier.first` will be nil by default.